app/validators/placeholders/placeholder_validator.py

```diff
@@ -144,19 +144,26 @@ def validate_answer_and_transform_unit_match(self, *, arguments, transform_type)
         answer_id = value.get("identifier")
         unit = arguments["unit"]
 
-        if self.errors or (
-            unit
-            == self.questionnaire_schema.answers_with_context[answer_id]["answer"][
-                "unit"
-            ]
-        ):
+        if value.get("source") == "calculated_summary":
+            if answer_ids := self.questionnaire_schema.get_calculation_block_ids(
+                block=self.questionnaire_schema.get_block(value.get("identifier")),
+                source_type="answers",
+            ):
+                core_answer_id = answer_ids[0]
+
+        else:
+            core_answer_id = value.get("identifier")
+
+        answers_unit = self.questionnaire_schema.answers_with_context[core_answer_id][
+            "answer"
+        ]["unit"]
+
+        if self.errors or (unit == answers_unit):
             return None
 
         self.add_error(
```

Comment:
Uploading Screen Recording 2023-06-19 at 11.36.20.mov…

Reply:
https://github.com/ONSdigital/eq-questionnaire-validator/assets/108286538/27c36514-2d68-43cb-a77c-726d2d297afb

Reply:
second time's the charm

Reply:
Ok, seems important to fix this so I've changed the logic and added extra tests so we know it works for all scenarios above now. Thanks!

Reply:
Nice one 👌